src/components/account/GetAccountId.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useState } from 'react'
+import getRequestStatus from 'utils/getRequestStatus'
+import validateAccountId from 'utils/validateAccountId'
+
+const GetAccountId = () => {
+  const [accountId, setAccountId] = useState('')
+  const [accountIdResult, setAccountIdResult] =
+    useState<AccountResponse | null>()
+
+  const [requestStatus, setRequestStatus] = useState<
+    'fetching' | 'error' | 'idle' | 'success'
+  >('idle')
+
+  const [errorMessage, setErrorMessage] = useState<string | null>(null)
+
+  useEffect(() => {
+    if (requestStatus !== 'success') setAccountIdResult(null)
+  }, [requestStatus])
+
+  return (
+    <div className="sectionContainer">
+      <h3>Search for an account</h3>
+      <input
+        type="text"
+        value={accountId}
+        placeholder="9e28507a-632f-44a6-99a7-98695cf2adcf"
+        onChange={(e) => setAccountId(e.target.value)}
+      ></input>
+      <button
+        className="getAccountIdButton"
+        onClick={() => {
+          const validateInput = validateAccountId(accountId)
+          if (validateInput) {
+            setErrorMessage(validateInput)
+            setRequestStatus('error')
+          } else {
+            setRequestStatus('fetching')
+            fetch(`https://nestjs-bank-app.herokuapp.com/accounts/${accountId}`)
+              .then((res) => {
+                if (!res.ok) {
+                  setRequestStatus('error')
+                  throw Error(res.statusText)
+                }
+                setRequestStatus('success')
+                return res.json()
+              })
+              .then((data) => setAccountIdResult(data))
+              .catch((e) => {
+                setErrorMessage(e.message)
+              })
+          }
+        }}
+      >
+        Search
+      </button>
+
+      <div className="requestStatus">
+        {getRequestStatus(requestStatus, errorMessage)}
```

Comment:
Feels like this function is just a slimmed down version of `postDeleteRequestStatus`. To prevent duplication just use one component and here you can wrap the component in a conditional to only render if `requestStatus === 'fetching' || requestStatus === 'error'` if you don't want anything shown for the other cases.

Reply:
One other thing is you will need to do for this component is make the `formErrors` an optional prop. This also makes me think it would be a good idea to consolidate all your `requestStatus`s to a single global type that doesn't have 'validationerrors'. Validation errors can be tracked solely in the separate useState that stores the validation error message array. 

It makes sense because really validation errors are unrelated to the request. They are (should be) triggered _before_ the request ever leaves the 'idle' state.